controller/deployer/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/bgentry/que-go"
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/jackc/pgx"
+	"github.com/flynn/flynn/controller/client"
+	"github.com/flynn/flynn/controller/deployer/strategies"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/postgres"
+	"github.com/flynn/flynn/pkg/shutdown"
+)
+
+type context struct {
+	db     *postgres.DB
+	client *controller.Client
+}
+
+func main() {
+	client, err := controller.NewClient("", os.Getenv("AUTH_KEY"))
+	if err != nil {
+		log.Fatalln("Unable to create controller client:", err)
+	}
+
+	if err := discoverd.Register("flynn-deployer", ":"+os.Getenv("PORT")); err != nil {
+		log.Fatal(err)
+	}
+
+	postgres.Wait("")
+	db, err := postgres.Open("", "")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	cxt := context{db: db, client: client}
+
+	pgxcfg, err := pgx.ParseURI(fmt.Sprintf("http://%s:%s@%s/%s", os.Getenv("PGUSER"), os.Getenv("PGPASSWORD"), db.Addr(), os.Getenv("PGDATABASE")))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	pgxpool, err := pgx.NewConnPool(pgx.ConnPoolConfig{
+		ConnConfig:   pgxcfg,
+		AfterConnect: que.PrepareStatements,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer pgxpool.Close()
+
+	q := que.NewClient(pgxpool)
+	wm := que.WorkMap{"Deployment": cxt.handleJob}
+
+	workers := que.NewWorkerPool(q, wm, 10)
+	go workers.Start()
+	shutdown.BeforeExit(func() { workers.Shutdown() })
+
+	<-make(chan bool) // block and keep running
```

Comment:
This should probably just call `workers.Start` directly rather than in a goroutine